### PR TITLE
use vim.fs.find instead vim.fn.findfile

### PR DESCRIPTION
**Pull request by fj0r**
it seems that `vim.fn.findfile` could not work (neovim 8.2)

use `vim.fs.find` instead it

### CODE REVIEW

**Comment by fj0r** [comment]
`vim.fn.findfile` always return empty string ""

**Comment by stevearc** [comment]
While I appreciate the proactive PR, I have some questions about the bug you're reporting. Are you saying that `:lua =vim.fn.findfile("justfile", "path/to/dir;")` is always returning an empty string? Even when there is a justfile in that or a parent directory? I haven't seen that behavior, and I'm on Neovim 0.8.2. Can you provide a minimal repro directory structure and script?

**Comment by fj0r** [comment]
Here is my [config](https://github.com/fj0r/nvim)

When I run `nvim -u NORC`, `:lua =vim.fn.findfile("justfile", "path/to/dir;")` works. With the configuration above, a empty character is always returned, no matter whether the target file exists or not. It seems to be a problem with my configuration file, I will study it again.

**Comment by fj0r** [comment]
When I set `:set wildignore=,.pyc,.swp` (comma first), `vim.fn.findfile` doesn't work. Then I set `.pyc,.swp` to be ok, it looks a bit weird.

If you use `set wildignore+=xxx`, there should be no problem. But I don't know how to set it in lua, now use `vim.o.wildignore = vim.o.wildignore .. ',xxx'`

**Comment by fj0r** [comment]
ok, i found it `vim.opt.wildignore:append { "*.pyc", "node_modules" }`

**Comment by fj0r** [comment]
By the way, I debugged for a long time, but couldn't find the root cause. When I run `OverseerRun`, the pop-up interface is not a telescope, but this
截图 2023-02-05 10-59-54
This is my [configuration](https://github.com/fj0r/nvim-lua/blob/main/lua/extensions/overseer.lua), is there something wrong with my configuration?

File: lua/overseer/template/just.lua [hunk 1]
```diff
@@ -1,15 +1,16 @@
 local log = require("overseer.log")
+local files = require("overseer.files")
 
 return {
   cache_key = function(opts)
-    return vim.fn.fnamemodify(vim.fn.findfile("justfile", opts.dir .. ";"), ":p")
+    return files.findfile('justfile', opts.dir, true)
```
Comment (stevearc) [inline lua/overseer/template/just.lua:6]: I don't think we need a wrapper function. You can just call `vim.fs.find("justfile", {path = opts.dir, type = "file", upward = true})[1]` directly. If you want you can extract that to a small helper function in this file so you don't have to repeat it on line 13.

File: lua/overseer/template/vscode/vs_util.lua [hunk 2]
```diff
@@ -18,11 +18,7 @@ end
 
 ---@param dir string
 M.get_tasks_file = function(dir)
-  local filename = vim.fn.findfile(files.join(".vscode", "tasks.json"), dir .. ";")
-  if filename ~= "" then
-    filename = vim.fn.fnamemodify(filename, ":p")
-  end
-  return filename
+  return files.findfile('tasks.json', files.join(dir, '.vscode'), true)
```
Comment (stevearc) [inline lua/overseer/template/vscode/vs_util.lua:21]: This is trickier. `vim.fn.findfile` allows searching upwards for a path like `.vscode/tasks.json`, but `vim.fs.find` does not. This change actually causes the VS Code task provider to not be able to find tasks if you search in a subdirectory of the root.

**Review by stevearc — CHANGES_REQUESTED** [review]
Hmmm...I forgot that `findfile()` respects wildignore. It's generally not a problem, but I guess it would be better to rely on `vim.fs.find`, which has a bit less magic going on.

That said if it's working for you now feel free to drop this. I'll migrate to using more `vim.fs.*` functions at some point.

You're not getting a telescope popup because overseer uses `vim.ui.select`. The default behavior is to use `inputlist()`, but you can patch that to use telescope instead. The easiest way to do that would be a plugin like [dressing.nvim](https://github.com/stevearc/dressing.nvim/) or [telescope-ui-select.nvim](https://github.com/nvim-telescope/telescope-ui-select.nvim)

**Comment by fj0r** [comment]
> That said if it's working for you now feel free to drop this. I'll migrate to using more vim.fs.* functions at some point.

it feels reasonable to keep the current code

> You're not getting a telescope popup because overseer uses vim.ui.select. The default behavior is to use inputlist(), but you can patch that to use telescope instead. The easiest way to do that would be a plugin like [dressing.nvim](https://github.com/stevearc/dressing.nvim/) or [telescope-ui-select.nvim](https://github.com/nvim-telescope/telescope-ui-select.nvim)

i will try. thank you

**Comment by stevearc** [comment]
Closing since I have since completed the migration to `vim.fs.find`

**Pull request closed without merging**